main.py

```diff
@@ -0,0 +1,99 @@
+import streamlit as sl
+from transformers import pipeline, set_seed
+
+
+def is_int(str):
+    try:
+        int(str)
+        return True
+    except ValueError:
+        return False
+
+
+@sl.cache(suppress_st_warning=True, allow_output_mutation=True)
+def init():
+    return pipeline('text-generation', model='gpt2')
+
+
+GENERATOR = init()
+
+
+def generate_text(input_text, seed, max_len, num_return_seq):
+    if is_int(seed):
+        seed = abs(int(seed))
+        if seed > 2 ** 32 - 1:
+            set_seed(42)
+        else:
+            set_seed(seed)
+    else:
+        set_seed(42)
+
+    if is_int(max_len):
+        max_len = abs(int(max_len))
+    else:
+        max_len = 30
+
+    if is_int(num_return_seq):
+        num_return_seq = abs(int(num_return_seq))
+    else:
+        num_return_seq = 5
+
+    return GENERATOR(input_text,
```

Comment:
This is another not-best solution. You define the model globally, but use it only inside this function 🤷‍♂️

Reply:
Also, there are much more parameters that you can specify and improve generation. You can read this article to understand different generation algorithms and their parameters (https://huggingface.co/blog/how-to-generate).

To handle these strategies, there is a special method [generate](https://huggingface.co/docs/transformers/v4.25.1/en/main_classes/text_generation#transformers.GenerationMixin.generate) in 🤗. Your pipeline will forward parameters into this function.

There are a lot of parameters, but you only need a couple of them:

- `max_length` and `min_length` to control the size of generated text (you already used it)
- `num_beams` to control the size of beam search, a bigger value gives more accurate, but strict text
- `do_sample` activate sampling strategies that add a little bit of randomness, these make the text more alive.
- `top_p` and `top_k` control the randomness. 

Let's also add these values into streamlit, or provide other UI to show differences between generations.